### PR TITLE
Add default value suggester in mitmweb options

### PR DESCRIPTION
**Pull request by MatthewShao**
From #2450 , `OptionDefault` component will compare current value and the default value, if they are not the same, it will show the default value.

### CODE REVIEW

**Comment by MatthewShao** [comment]
Here's a simple demo for this new feature:
peek 2017-07-22 21-18

**Comment by MatthewShao** [comment]
I made the config file write to `~/.mitmproxy/config.yaml`, which is identical to the path in the console. So when we update the options in `mitmweb` and then switch to `mitmproxy`, the dumped options can be loaded automatically.

File: mitmproxy/tools/web/app.py [hunk 1]
```diff
@@ -18,8 +18,11 @@
 from mitmproxy import log
 from mitmproxy import version
 from mitmproxy import optmanager
+from mitmproxy import options
 import mitmproxy.tools.web.master # noqa
 
+CONFIG_PATH = os.path.join(options.CA_DIR, 'config.yaml')
```
Comment (mhils) [inline mitmproxy/tools/web/app.py]: Don't redefine, just import the existing one in `cmdline.py`. :)

File: mitmproxy/tools/web/app.py [hunk 2]
```diff
@@ -475,7 +486,8 @@ def __init__(self, master, debug):
                 FlowContentView),
             (r"/settings", Settings),
             (r"/clear", ClearAll),
-            (r"/options", Options)
+            (r"/options", Options),
+            (r"/options/dump", DumpOptions)
```
Comment (mhils) [inline mitmproxy/tools/web/app.py]: Really no strong opinion, just curious - why "dump" and not "save"?

File: mitmproxy/tools/web/master.py [hunk 3]
```diff
@@ -125,6 +125,12 @@ def run(self):  # pragma: no cover
                     "No web browser found. Please open a browser and point it to {}".format(web_url),
                     "info"
                 )
+        unknown_opts = optmanager.load_paths(app.CONFIG_PATH)
```
Comment (mhils) [inline mitmproxy/tools/web/master.py]: This should already happen by default [here](https://github.com/mitmproxy/mitmproxy/blob/8526ca9e175af4a903348c39928535ae10f7543d/mitmproxy/tools/main.py#L69) if I understand things correctly, is that not the case?

File: web/src/js/__tests__/ducks/optionsSpec.js [hunk 4]
```diff
@@ -49,3 +49,18 @@ describe('sendUpdate', () => {
         ])
     })
 })
+
+describe('dump', () => {
```
Comment (mhils) [inline web/src/js/__tests__/ducks/optionsSpec.js]: Pressing Escape does not save the file now, does it? I was also thinking about making the modal disappear by clicking on the background. Maybe we actually want to do this in `hideModal`, or OptionModal's `componentWillUnmount`? I totally see the motivation why you wanted to separate this, but that looks it leaves a lot of room for error.

File: mitmproxy/tools/web/app.py (hunk 2 above)
Comment (MatthewShao) [inline mitmproxy/tools/web/app.py]: I followed the pattern of `DumpFlows` befoe, but seems `SaveOptions` sounds more reasonable here.

File: mitmproxy/tools/web/master.py (hunk 3 above)
Comment (MatthewShao) [inline mitmproxy/tools/web/master.py]: Did not notice that... This should be removed.

**Pull request merged**